lib/_http_server.js

```diff
@@ -295,6 +296,24 @@ ServerResponse.prototype.writeProcessing = function writeProcessing(cb) {
   this._writeRaw('HTTP/1.1 102 Processing\r\n\r\n', 'ascii', cb);
 };
 
+ServerResponse.prototype.writeEarlyHints = function writeEarlyHints(links, cb) {
+  let head = 'HTTP/1.1 103 Early Hints\r\n';
+
+  if (typeof links === 'string') {
+    validateLinkHeaderValue(links, 'links');
+    head += 'Link: ' + links + '\r\n';
+  } else if (ArrayIsArray(links)) {
+    links.forEach((link) =>
+      validateLinkHeaderValue(link, 'links')
+    );
+    head += 'Link: ' + links.join(',') + '\r\n';
```

Comment:
I think this could be improved a bit by removing the double iteration.

```suggestion
    head += 'Link: '
    for (let i = 0; i < links.length; i++) {
      const link = links[i]
      validateLinkHeaderValue(link, 'links')
      head += link
      if (i > 0) {
        head += ','
      }
    }
    head += '\r\n'
```

Reply:
I'll add your suggestion but I'll tweak a little so it complies with the linting rules and change the condition for appending the comma for every item unless it's the last one.
```js
    head += 'Link: '

    for (let i = 0; i < links.length; i++) {
      const link = links[i];
      validateLinkHeaderValue(link, 'links');
      head += link;

      if (i !== links.length -1) {
        head += ', ';
      }
    }

    head += '\r\n';
```